biogrid/libs/biogrid-simulator/src/lib/biogrid/biogrid-class.ts

```diff
@@ -209,11 +254,105 @@ export class Biogrid implements Grid {
     const rows = Math.ceil(numberOfGridItems / cols);
     const positions: ItemPosition[] = [];
     for (let i = 0; i < numberOfGridItems; i++) {
-      positions.push({
-        x: (((i % cols) + 0.5) / cols) * townSize.width,
-        y: ((Math.floor(i / cols) + 0.5) / rows) * townSize.height,
-      });
+      const newPositionUnverified = {
+        x: this.roundToGridDistance(
+          (((i % cols) + 0.5) / cols) * townSize.width
+        ),
+        y: this.roundToGridDistance(
+          ((Math.floor(i / cols) + 0.5) / rows) * townSize.height
+        ),
+      };
+      const newPosition = this.findNearestUnoccupiedPosition(
+        newPositionUnverified,
+        townSize
+      );
+      positions.push(newPosition);
+      this.itemInPosition[this.formatItemPosition(newPosition)] = true;
     }
     return positions;
   }
+
+  /**
+   * Find the nearest unoccupied position to {@code pos} by looking looking in a spiral with pos at its center
+   * First the space immediatly right of pos is checked, then the one above, then to the left, then below, then two right spaces out, two right up, etc
+   */
+  private findNearestUnoccupiedPosition(
+    pos: ItemPosition,
+    townSize: TownSize
+  ): ItemPosition {
+    if (!this.positionOccupied(pos)) {
+      return pos;
+    }
+    let radius = GRID_DISTANCES.INCREMENTS_KM;
+    let angle = 0;
+    let outOfBoundsCount = 0;
+    let xOffset = 0,
+      yOffset = 0;
+    let newPos = { x: pos.x + xOffset, y: pos.y + yOffset };
```

Comment:
since xoffset and y offset start off as 0, won't this check the exact same position you already checked in line 283?

Reply:
Oh yeah it will. I removed the if condition